// A tensor library for Hermes

enum DataType {
  DTYPE_INT = 1;
  DTYPE_LONG = 2;
  DTYPE_FLOAT = 3;
  DTYPE_DOUBLE = 4;
}

message Tensor {
  repeated uint64 dims = 1;
  optional DataType dtype = 2;
  optional bytes data = 3;
}

message BlockDistTensor {
  optional Tensor objrefs = 1; // tensor of objrefs, each representing a block subtensor living on a node
}
